From 6ff4cd1160afafc12ad1603e3d2f39256e4b708d Mon Sep 17 00:00:00 2001
From: Hongbo Li <lihongbo22@huawei.com>
Date: Tue, 27 Aug 2024 10:45:15 +0800
Subject: [PATCH] lib/string_choices: Add str_true_false()/str_false_true()
 helper
Git-commit: 6ff4cd1160afafc12ad1603e3d2f39256e4b708d
References: git-fixes
Patch-mainline: v6.12-rc1

Add str_true_false()/str_false_true() helper to return "true" or
"false" string literal.

Signed-off-by: Hongbo Li <lihongbo22@huawei.com>
Link: https://lore.kernel.org/r/20240827024517.914100-2-lihongbo22@huawei.com
Signed-off-by: Kees Cook <kees@kernel.org>
Signed-off-by: Oliver Neukum <oneukum@suse.com>

---
 include/linux/string_choices.h |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/include/linux/string_choices.h
+++ b/include/linux/string_choices.h
@@ -42,6 +42,12 @@ static inline const char *str_yes_no(boo
 	return v ? "yes" : "no";
 }
 
+static inline const char *str_true_false(bool v)
+{
+	return v ? "true" : "false";
+}
+#define str_false_true(v)		str_true_false(!(v))
+
 /**
  * str_plural - Return the simple pluralization based on English counts
  * @num: Number used for deciding pluralization
